code/modules/projectiles/guns/shotguns.dm

```diff
@@ -1049,8 +1053,44 @@ can cause issues with ammo types getting mixed up during the burst.
 
 /obj/item/weapon/gun/shotgun/pump/cmb/Initialize(mapload, spawn_empty)
 	. = ..()
+	primary_tube = current_mag
+	secondary_tube = new current_mag.type(src, spawn_empty? 1:0)
+	current_mag = secondary_tube
+	replace_tube(current_mag.current_rounds)
+	active_tube = current_mag
 	pump_delay = FIRE_DELAY_TIER_5*2
 
+/obj/item/weapon/gun/shotgun/pump/cmb/Destroy()
+	QDEL_NULL(primary_tube)
+	QDEL_NULL(secondary_tube)
+	QDEL_NULL(active_tube)
+	. = ..()
+
+/obj/item/weapon/gun/shotgun/pump/cmb/proc/do_toggle_tube(var/mob/user)
+	if(!ishuman(user) || user.is_mob_incapacitated())
+		return FALSE
+	var/obj/item/weapon/gun/shotgun/pump/cmb/shotgun = user.get_active_hand()
+	if(!istype(shotgun))
+		to_chat(usr, "You must be holding [src] to change the internal tube magazine")
+		return
+	if(!active_tube)
+		return
+	if(active_tube == primary_tube)
+		active_tube = secondary_tube
+	else
+		active_tube = primary_tube
+	to_chat(user, "You switch the internal tube magazine.")
+	playsound(src, 'sound/machines/click.ogg', 20, 1)
+	shotgun.current_mag = active_tube
+
+	return TRUE
+
+/obj/item/weapon/gun/shotgun/pump/cmb/verb/toggle_tube()
```

Comment:
really don't think it should be called 'toggle', to toggle something is to turn it on or off. this is just swapping between identical tubes. Change to 'swap'?